modules/lang-painless/src/main/resources/org/elasticsearch/painless/spi/org.elasticsearch.script.fields.txt

```diff
@@ -13,7 +13,41 @@ class org.elasticsearch.script.Field {
   String getName()
   boolean isEmpty()
   List getValues()
-  def getValue(def)
+
+  org.elasticsearch.script.LongField asLongField()
+  long asLong(long)
+
+  org.elasticsearch.script.DoubleField asDoubleField()
+  double asDouble(double)
+
+  org.elasticsearch.script.BigIntegerField asBigIntegerField()
+  BigInteger asBigInteger(BigInteger)
+
+  org.elasticsearch.script.StringField asStringField()
+  String asString(String)
+
+  org.elasticsearch.script.ObjectField asObjectField()
+  Object asObject(Object)
+}
+
+class org.elasticsearch.script.LongField {
+  long getValue(long)
+}
+
+class org.elasticsearch.script.DoubleField {
+  double getValue(double)
+}
+
+class org.elasticsearch.script.BigIntegerField {
+  BigInteger getValue(BigInteger)
+}
+
+class org.elasticsearch.script.StringField {
+  String getValue(String)
+}
+
+class org.elasticsearch.script.ObjectField {
```

Comment:
Please change this from Object to def to support duck-typing as opposed to just the base Java Object which doesn't make sense to cover most extraneous cases.

Reply:
Done.